pkg/config/instrumentation/config_metrics.go

```diff
@@ -0,0 +1,59 @@
+package instrumentation
+
+import (
+	"crypto/sha256"
+	"fmt"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+type configMetrics struct {
+	configHash        *prometheus.GaugeVec
+	configLoadSuccess prometheus.Gauge
+	configLoadSeconds *prometheus.GaugeVec
+}
+
+// ConfigMetrics exposes metrics related to configuration loading
+var ConfigMetrics = newConfigMetrics()
+
+func newConfigMetrics() *configMetrics {
+	var m configMetrics
+
+	m.configHash = promauto.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "agent_config_hash",
+			Help: "Hash of the currently active config file.",
+		},
+		[]string{"sha256"},
+	)
+	m.configLoadSuccess = promauto.NewGauge(prometheus.GaugeOpts{
+		Name: "agent_config_last_load_successful",
+		Help: "Config loaded successfully.",
+	})
+	m.configLoadSeconds = promauto.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "agent_config_last_load_timestamp_seconds",
+		Help: "Timestamp of the last configuration load by result.",
+	}, []string{"result"})
```

Comment:
Hm, one of my concerns with the `result` label is that every instance of the label will continue to be reported for the process lifetime, which might be confusing and hard to query for exactly what you're looking for. 

I think there's two choices here:

1. Remove the result label 
2. Delete all the labels on the vector every time we instrument a new load, so the /metrics endpoint only reports the metric and label corresponding to the most recent load.  

Personally, I'd go for the first; a user can still know when unsuccessful loads happened by querying for `agent_config_last_load_timestamp_seconds and agent_config_last_load_successful == 0`.

Reply:
cc @mattdurham since I'm suggesting that the `result` label might cause issues

Reply:
Seems reasonable.

Reply:
SGTM 👍